ncbi/src/main/java/org/reactome/release/UCSC.java

```diff
@@ -0,0 +1,231 @@
+			System.lineSeparator() + System.lineSeparator();
+	}
+
+	/**
+	 * Retrieves Map of Reactome UniProt instance to lines for UCSC Event File for that UniProt instance
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Map of UniProt Reactome Entry to Set of Strings containing UCSC Event Lines
+	 */
+	private Map<UniProtReactomeEntry, Set<String>> getUniProtReactomeEntriesToUCSCEventLines(Session graphDBSession) {
+		return getUniProtReactomeEntriesForUCSC(graphDBSession)
+			.stream()
+			.collect(Collectors.toMap(
+				uniProtReactomeEntry -> uniProtReactomeEntry,
+				uniProtReactomeEntry -> getUCSCEventLines(uniProtReactomeEntry, graphDBSession),
+				(oldValue, newValue) -> newValue,
+				TreeMap::new
+			));
+	}
+
+	/**
+	 * Retrieves lines, describing UniProt to Reactome Event relationships (including event stable identifier and
+	 * display name), to include in the UCSC Event file for a given UniProt instance
+	 * @param uniProtReactomeEntry UniProt instance in Reactome
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Set of UniProt Reactome Entry objects
+	 */
+	private Set<String> getUCSCEventLines(UniProtReactomeEntry uniProtReactomeEntry, Session graphDBSession) {
+		return uniProtReactomeEntry
+			.getEvents(graphDBSession)
+			.stream()
+			.map(event ->
+				String.join(
+					"\t",
+					uniProtReactomeEntry.getAccession(),
+					event.getStableIdentifier(),
+					event.getName()
+				)
+			)
+			.collect(Collectors.toCollection(LinkedHashSet::new));
+	}
+
+	/**
+	 * Generates error message when a UniProt instance has no Reactome Events found in which it participates
+	 * @param uniProtReactomeEntry UniProt instance in Reactome
+	 * @return Error message for UniProt instance
+	 */
+	private String getNoEventsErrorMessage(UniProtReactomeEntry uniProtReactomeEntry) {
+		return uniProtReactomeEntry.getDisplayName() +
+			" participates in Event(s) but no top Pathway can be found, " +
+			" i.e. there seem to be a pathway" +
+			" which contains or is an instance of itself.";
+	}
+
+	/**
+	 * Retrieves all UniProt instances in Reactome with an EWAS of species human, rat, or mouse
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Set of UniProt Reactome Entry objects
+	 */
+	Set<UniProtReactomeEntry> getUniProtReactomeEntriesForUCSC(Session graphDBSession) {
+		if (ucscUniProtReactomeEntries != null) {
+			return ucscUniProtReactomeEntries;
+		}
+
+		logger.info("Fetching UniProt Reactome Entries for UCSC");
+
+		ucscUniProtReactomeEntries = graphDBSession.run(
+			String.join(System.lineSeparator(),
+				"MATCH (ewas:EntityWithAccessionedSequence)-[:referenceEntity]->(rgp:ReferenceGeneProduct)" +
+				"-[:referenceDatabase]->(rd:ReferenceDatabase)",
+				"WHERE ewas.speciesName IN ['Homo sapiens', 'Rattus norvegicus', 'Mus musculus'] AND rd.displayName =" +
```

Comment:
What's special about these three species?

Reply:
I'm not sure, to be honest.  This has been in the UCSC data export since I've started with Reactome.  I can ask Robin and Peter if this is still correct.

Reply:
Ok.